src/VisualStudio/TestUtilities/IntegrationHost.cs

```diff
@@ -0,0 +1,522 @@
+            // TODO: Reset Roslyn Options
+
+            // TODO: Enable Tracking of Active Token
+            // TODO: Disable Partial Solutions
+
+        }
+
+        internal void ExecuteDteCommand(string command, string args = "", int timeout = IntegrationHelper.DefaultTimeout)
+        {
+            WaitForDteCommandAvailability(command, timeout);
+            _dte.ExecuteCommand(command, args);
+        }
+
+        internal AutomationElement LocateAutomationElement(string elementName, bool recursive = false, int timeout = IntegrationHelper.DefaultTimeout)
+        {
+            var automationElement = null as AutomationElement;
+            var scope = (recursive ? TreeScope.Descendants : TreeScope.Children);
+            var condition = new PropertyCondition(AutomationElement.NameProperty, elementName);
+
+            IntegrationHelper.WaitForResult(() => {
+                automationElement = AutomationElement.RootElement.FindFirst(scope, condition);
+                return (automationElement != null);
+            }, expectedResult: true, timeout: timeout);
+
+            return automationElement;
+        }
+
+        internal Window LocateDteWindow(string windowTitle, int timeout = IntegrationHelper.DefaultTimeout)
+        {
+            var dteWindow = null as Window;
+
+            IntegrationHelper.WaitForResult(() => {
+                foreach (Window window in _dte.Windows)
+                {
+                    if (window.Caption.Equals(windowTitle, StringComparison.OrdinalIgnoreCase))
+                    {
+                        dteWindow = window;
+                        return true;
+                    }
+                }
+                return false;
+            }, expectedResult: true, timeout: timeout);
+
+            return dteWindow;
+        }
+
+        internal T ExecuteOnHostProcess<T>(Type type, string methodName, BindingFlags bindingFlags, params object[] parameters)
+             => ExecuteOnHostProcess<T>(type.Assembly.Location, type.FullName, methodName, bindingFlags, parameters);
+
+        internal T ExecuteOnHostProcess<T>(string assemblyFilePath, string typeFullName, string methodName, BindingFlags bindingFlags, params object[] parameters)
+        {
+            var objectUri = _service.Execute(assemblyFilePath, typeFullName, methodName, bindingFlags, parameters);
+
+            if (string.IsNullOrWhiteSpace(objectUri))
+            {
+                return default(T);
+            }
+
+            return (T)(Activator.GetObject(typeof(T), $"{_serviceUri}/{objectUri}"));
+        }
+
+        internal void WaitForDteCommandAvailability(string command, int timeout = IntegrationHelper.DefaultTimeout)
+            => IntegrationHelper.WaitForResult(() => Dte.Commands.Item(command).IsAvailable, expectedResult: true, timeout: timeout);
+
+        protected virtual void Dispose(bool disposing)
+        {
+            _requireNewInstance |= (!disposing);
```

Comment:
I find this pattern to be odd.  Either this test required a new instance or it didn't.  Changing this value in `Dispose` means the validity of this flag depends on the time in which I'm looking at it.  Perhaps this flag should be better named or made an explicit parameter to `Cleanup`